File: Models/Route.cs
namespace RouteMK.Models;

public class Route
{
    public int RouteId { get; set; }

    public int TransportOrganizerId { get; set; }

    public int FromLocationId { get; set; }

    public int ToLocationId { get; set; }

    public virtual ICollection<Favorite> Favorites { get; set; } = new List<Favorite>();

    public virtual Location FromLocation { get; set; } = null!;

    public virtual Location ToLocation { get; set; } = null!;

    public virtual Account TransportOrganizer { get; set; } = null!;

    public virtual ICollection<TripDaysActive> TripDaysActives { get; set; } = new List<TripDaysActive>();

    public virtual ICollection<Trip> Trips { get; set; } = new List<Trip>();
}

File: src/main/java/mk/route/routemk/models/Route.java
package mk.route.routemk.models;

import jakarta.persistence.*;

@Entity
@Table(name = "route")
public class Route {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "route_id")
    private Integer routeId;

    @ManyToOne
    @JoinColumn(name = "from_location_id", referencedColumnName = "location_id")
    private Location source;

    @ManyToOne
    @JoinColumn(name = "to_location_id", referencedColumnName = "location_id")
    private Location destination;

    @ManyToOne
    @JoinColumn(name = "transport_organizer_id", referencedColumnName = "transport_organizer_id")
    private TransportOrganizer tranOrg;

    public Route(Location source, Location destination, TransportOrganizer tranOrg) {
        this.source = source;
        this.destination = destination;
        this.tranOrg = tranOrg;
    }

    public TransportOrganizer getTranOrg() {
        return tranOrg;
    }

    public Route() {

    }
}
